src/fr/anisama/src/eu/kanade/tachiyomi/animeextension/fr/anisama/AniSama.kt

```diff
@@ -0,0 +1,345 @@
+
+    override fun latestUpdatesNextPageSelector() = popularAnimeNextPageSelector()
+
+    override fun latestUpdatesParse(response: Response): AnimesPage {
+        aniSamaFilters.fetchFilters()
+        return super.latestUpdatesParse(response)
+    }
+
+    // =============================== Search ===============================
+    override fun getFilterList() = aniSamaFilters.getFilterList()
+
+    override suspend fun getSearchAnime(page: Int, query: String, filters: AnimeFilterList): AnimesPage {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val id = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/anime/$id")).awaitSuccess().use(::searchAnimeByIdParse)
+        } else {
+            super.getSearchAnime(page, query, filters)
+        }
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response.asJsoup())
+            .apply { setUrlWithoutDomain(response.request.url.toString()) }
+        return AnimesPage(listOf(details), false)
+    }
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val url = "$baseUrl/filter".toHttpUrl().newBuilder()
+        url.addQueryParameter("keyword", query)
+        url.addQueryParameter("page", page.toString())
+        filters.filterIsInstance<AniSamaFilters.QueryParameterFilter>().forEach {
+            val (name, value) = it.toQueryParameter()
+            if (value != null) url.addQueryParameter(name, value)
+        }
+        return GET(url.build())
+    }
+
+    override fun searchAnimeSelector() = popularAnimeSelector()
+
+    override fun searchAnimeFromElement(element: Element) = popularAnimeFromElement(element)
+
+    override fun searchAnimeNextPageSelector() = popularAnimeNextPageSelector()
+
+    // =========================== Anime Details ============================
+    private fun Elements.getMeta(name: String) = select(".item:has(.item-title:contains($name)) > .item-content")
+
+    private fun parseStatus(status: String) = when (status) {
+        "Terminer" -> SAnime.COMPLETED
+        "En cours" -> SAnime.ONGOING
+        else -> SAnime.UNKNOWN
+    }
+
+    override fun animeDetailsParse(document: Document): SAnime {
+        val details = document.select(".anime-detail")
+        return SAnime.create().apply {
+            title = details.select(".dynamic-name").text().substringBeforeLast(" ")
+            thumbnail_url = details.select(".film-poster-img").attr("src")
+            url = document.select("link[rel=canonical]").attr("href")
+            artist = details.getMeta("Studio").text()
+            status = parseStatus(details.getMeta("Status").text())
+            description = details.select(".shorting").text()
+            genre = details.getMeta("Genre").text()
+            initialized = true
```

Comment:
This line isn't necessary (unless you're overriding `getMangaDetails`, which you aren't)